src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs

```diff
@@ -0,0 +1,201 @@
+
+            Document doc = context.Document;
+
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (root.FindNode(context.Span, getInnermostNodeForTie: true) is not SyntaxNode node)
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (model.GetOperation(node, ct) is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison)
+                is not INamedTypeSymbol stringComparisonType)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparer)
+                is not INamedTypeSymbol stringComparerType)
+            {
+                return;
+            }
+
+            // Only diagnose overloads that take one string argument:
+            // Contains(string), IndexOf(string), StartsWith(string), CompareTo(string)
+            if (invocation.Arguments.Count() != 1)
+            {
+                return;
+            }
+
+            // There should be a child invocation on the left side (instance) of the large invocation
+            // If the large invocation is "a.ToLower().Contains(b)"
+            // Its instance is "a.ToLower()", the child invocation
+            if (invocation.Instance is not IInvocationOperation instanceOperation)
+            {
+                return;
+            }
+
+            string caseChangingMethodName;
+            switch (instanceOperation.TargetMethod.Name)
+            {
+                case RecommendCaseInsensitiveStringComparisonAnalyzer.StringToLowerMethodName or RecommendCaseInsensitiveStringComparisonAnalyzer.StringToUpperMethodName:
+                    caseChangingMethodName = RecommendCaseInsensitiveStringComparisonAnalyzer.StringComparisonCurrentCultureIgnoreCaseName;
+                    break;
+                case RecommendCaseInsensitiveStringComparisonAnalyzer.StringToLowerInvariantMethodName or RecommendCaseInsensitiveStringComparisonAnalyzer.StringToUpperInvariantMethodName:
+                    caseChangingMethodName = RecommendCaseInsensitiveStringComparisonAnalyzer.StringComparisonInvariantCultureIgnoreCaseName;
+                    break;
+                default:
+                    return;
+            }
+
+            Task<Document> createChangedDocument(CancellationToken _) => FixInvocationAsync(doc, root,
+                invocation, instanceOperation,
+                stringComparisonType, stringComparerType,
+                invocation.TargetMethod.Name, caseChangingMethodName, ct);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: MicrosoftNetCoreAnalyzersResources.RecommendCaseInsensitiveStringComparisonTitle,
```

Comment:
I wouldn't use the analyzer title as the codefix title, and would make the codefix title a bit shorter.

Reply:
I welcome suggestions. What would you use?

Reply:
Finding good titles is hard 😄 
Maybe:

- `Use '{0}' with 'StringComparison.{1}'` where `{0}` will be `Contains`, `IndexOf`, or `StartsWith`
- `Use 'StringComparer.{0}.Compare'

Reply:
Just to confirm: You're not suggesting that I change the actual resource, you're suggesting that I literally use the string you provided as the value for the title argument. Is this accurate?

Reply:
Sorry, nevermind. I thought I was typing on your equivalenceKey suggestion. Ignore.